subprojects/parseq/src/main/java/com/linkedin/parseq/Task.java

```diff
@@ -359,6 +359,40 @@
     return Task.withSideEffect("withSideEffect", func);
   }
 
+  /**
+   * Equivalent to {@code withSideEffect(desc, func)} in how the task will run and how return values are handled.
+   *
+   * Differs in exception handling behavior from {@code withSideEffect(desc, func)}.
+   * If an exception is thrown in {@link Function1}(passed in arguments) then {@code safeSideEffect } will convert it
+   * into a {@code Task.failure } and will ensure that the exception doesn't spread it's impact outside the plan
+   * compared to {@code withSideEffect} in which an exception is thrown in {@link Function1}, the entire plan fails.
+   *
+   * @param desc description of the side Effect. This will show up in trace
+   * @param func function to be applied to get side effect task
+   * @return a new Task that will run the side effect Task
+   */
+  default Task<T> safeSideEffect(final String desc, final Function1<? super T, Task<?>> func) {
```

Comment:
There are two versions of `withSideEffect` method, one taking `Function1`, the other taking `Callable`. We may also need to guard for `Callable` version.